hi-ml-azure/src/health_azure/utils.py

```diff
@@ -721,21 +730,38 @@ def get_workspace(aml_workspace: Optional[Workspace] = None, workspace_config_pa
         return aml_workspace
 
     if workspace_config_path is None:
+        logging.info(f"Trying to locate the workspace config file '{WORKSPACE_CONFIG_JSON}' in the current folder "
+                     "and its parent folders")
         workspace_config_path = find_file_in_parent_to_pythonpath(WORKSPACE_CONFIG_JSON)
         if workspace_config_path:
             logging.info(f"Using the workspace config file {str(workspace_config_path.absolute())}")
-        else:
-            raise ValueError("No workspace config file given, nor can we find one.")
 
-    if not isinstance(workspace_config_path, Path):
-        raise ValueError("Workspace config path is not a path, check your input.")
-    elif workspace_config_path.is_file():
-        auth = get_authentication()
+    workspace_name = None
+    subscription_id = None
+    resource_group = None
+    if workspace_config_path is None:
+        logging.info("Trying to load the environment variables that define the workspace.")
+        workspace_name = get_secret_from_environment(ENV_WORKSPACE_NAME, allow_missing=True)
+        subscription_id = get_secret_from_environment(ENV_SUBSCRIPTION_ID, allow_missing=True)
+        resource_group = get_secret_from_environment(ENV_RESOURCE_GROUP, allow_missing=True)
+
+    if workspace_config_path is not None and not workspace_config_path.is_file():
+        raise FileNotFoundError(f"Workspace config file does not exist: {workspace_config_path}")
+
+    auth = get_authentication()
+    # Check if all 3 environment variables are set
+    if bool(workspace_name) and bool(subscription_id) and bool(resource_group):
```

Comment:
[Minor] can we switch the order of 753 and 757 to keep the logic as described in doc string? is it meant to be sequential steps?

Reply:
good spot, thanks!